src/routes/conversation/[id]/upload-pdf/+server.ts

```diff
@@ -0,0 +1,41 @@
+import { authCondition } from "$lib/server/auth";
+import { collections } from "$lib/server/database";
+import { MAX_SEQ_LEN as CHUNK_CAR_LEN, createEmbeddings } from "$lib/server/embeddings";
+import { uploadPdfEmbeddings } from "$lib/server/files/uploadFile";
+import { chunk } from "$lib/utils/chunk";
+import { error } from "@sveltejs/kit";
+import { ObjectId } from "mongodb";
+import * as pdfjsLib from "pdfjs-dist/legacy/build/pdf";
+
+export async function POST({ request, params, locals }) {
+	const conversationId = new ObjectId(params.id);
+	const conversation = await collections.conversations.findOne({
+		_id: conversationId,
+		...authCondition(locals),
+	});
+
+	if (!conversation) {
+		throw error(404, "Conversation not found");
+	}
+
+	const formData = await request.formData();
+	const file = formData.get("pdf"); // 'pdf' is the name used in FormData on the frontend
+	const data = new Uint8Array(await file.arrayBuffer());
+	const loadingTask = pdfjsLib.getDocument({ data });
+	const pdf = await loadingTask.promise;
+
+	const N_MAX_PAGES = 20;
```

Comment:
This seems a bit low, 100 or 200 pages maybe?

Reply:
I see. 100 or 200 would be bit slow to create embeddings on CPU using the current solution of transformers.js (I will provide actual benchmark numbers). 

In this case, should I review/push for the community PR https://github.com/huggingface/chat-ui/pull/646 that makes it possible to have an embedding endpoint for faster embeddings creation. We can still use tfjs embeddings for websearch, and use TEI-powered embeddings endpoint for PDF & possible for assistants #639 (if users can upload documents). Wdyt ?